### PR TITLE
thrift: configurable validate_clusters to override the default cluste…

### PR DESCRIPTION
**Pull request by JuniorHsu**
…r validation

Signed-off-by: kuochunghsu <kuochunghsu@pinterest.com>

Commit Message:
This is a continuation of #20577.

Additional Description:
Risk Level: low
Testing: unit test
Docs Changes: proto
Release Notes: current.rst

### CODE REVIEW

**Comment by JuniorHsu** [comment]
cc  @rgs1 @tkovacs-2

File: test/extensions/filters/network/thrift_proxy/conn_manager_test.cc [hunk 1]
```diff
@@ -1960,6 +1960,28 @@ stat_prefix: test
                           "unknown thrift cluster");
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownClusterWithNegativeValidateClusters) {
```
Comment (rgs1) [inline test/extensions/filters/network/thrift_proxy/conn_manager_test.cc:1963]: nit: `UnknownClusterWithValidateClustersDisabled`

File: test/extensions/filters/network/thrift_proxy/conn_manager_test.cc [hunk 2]
```diff
@@ -1984,6 +2006,27 @@ TEST_F(ThriftConnectionManagerTest, UnknownWeightedCluster) {
                           "unknown thrift weighted cluster");
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownWeightedClusterWithNegativeValidateClusters) {
```
Comment (rgs1) [inline test/extensions/filters/network/thrift_proxy/conn_manager_test.cc:2009]: nit: `UnknownWeightedClusterWithValidateClustersDisabled`

File: test/extensions/filters/network/thrift_proxy/conn_manager_test.cc [hunk 3]
```diff
@@ -2028,6 +2071,25 @@ stat_prefix: test
                           "unknown thrift shadow cluster");
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownMirrorPolicyClusterWithNegativeValidateClusters) {
```
Comment (rgs1) [inline test/extensions/filters/network/thrift_proxy/conn_manager_test.cc:2074]: nit: `UnknownMirrorPolicyClusterWithValidateClustersDisabled`

**Review by rgs1 — CHANGES_REQUESTED** [review]
LGTM, thanks! Just a few nits.

**Review by rgs1 — DISMISSED** [review]
LGTM, thanks!

Over to @zuercher for merging.

File: api/envoy/extensions/filters/network/thrift_proxy/v3/route.proto [hunk 4]
```diff
@@ -30,6 +30,21 @@ message RouteConfiguration {
...
+  // load. If set to false and a route refers to a non-existent cluster, the
+  // route table will load and the router filter will return a INTERNAL_ERROR
+  // if the route is selected at runtime. This setting defaults to true if the route table
+  // is statically defined via the :ref:`route_config
+  // <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.route_config>`
+  // option. This setting default to false if the route table is loaded dynamically via the
+  // :ref:`trds
+  // <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.trds>`
+  // option. Users may wish to override the default behavior in certain cases (for example when
+  // using CDS with a static route table).
```
Comment (htuch) [inline api/envoy/extensions/filters/network/thrift_proxy/v3/route.proto:46]: Nit: not CDS

**Review by htuch — COMMENTED** [review]
API LGTM modulo nit.

File: api/envoy/extensions/filters/network/thrift_proxy/v3/route.proto [hunk 5]
```diff
@@ -30,6 +30,21 @@ message RouteConfiguration {
...
+  // load. If set to false and a route refers to a non-existent cluster, the
+  // route table will load and the router filter will return a INTERNAL_ERROR
+  // if the route is selected at runtime. This setting defaults to true if the route table
+  // is statically defined via the :ref:`route_config
+  // <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.route_config>`
+  // option. This setting default to false if the route table is loaded dynamically via the
+  // :ref:`trds
+  // <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.trds>`
+  // option. Users may wish to override the default behavior in certain cases (for example when
+  // using TRDS with a static route table).
```
Comment (tkovacs-2) [inline api/envoy/extensions/filters/network/thrift_proxy/v3/route.proto:46]: @htuch 
What was wrong with the CDS in this comment?
In this form it is meaningless, TRDS means dynamic routing table.

Comment (htuch) [reply api/envoy/extensions/filters/network/thrift_proxy/v3/route.proto:46]: Sorry, you're right, let's revert.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Review by zuercher — APPROVED** [review]
I went ahead and resolved the trivial merge conflict. I'd hoped that would revoke all the previous approvals, but it seems to have.

**Pull request merged**